src/utils/optimizeCallback.js

```diff
@@ -42,7 +55,21 @@ function _findSingleSignatureWithArity (callback, arity) {
   }
 }
 
-function _findNumberOfArguments (callback, value, index, array) {
+function _findIfCallbackIsUnary (callback) {
+  const callbackStr = callback.toString()
+  const firstParenthesisIndex = callbackStr.indexOf('(')
+  const firstClosingParenthesisIndex = callbackStr.indexOf(')', firstParenthesisIndex)
+  const paramsStr = callbackStr.slice(firstParenthesisIndex + 1, firstClosingParenthesisIndex)
```

Comment:
I think you can use a regex here too to extract the params:

```js
const paramsStr = callbackStr.toString().match(/\(.*?\)/)
``` 

(It includes the parenthesis but that is no problem: it is only used to test whether it contains a spread operator)